src/System.Text.Json/docs/writable_json_dom_spec.md

```diff
@@ -0,0 +1,257 @@
+    | 2 | - only one additional method <br> - accepts collection of different types <br > | - works  only in C# <br> - no IntelliSense <br> - users may not be aware of it <br> - accepts only `Array` <br> - accepts collection of types not deriving from `JsonNode` <br> - needs to check it in runtime | from {1,2}, <br>2 seems worse |
+    | 3 | - accepts IEnumerable <br> - does not accept collection of  types not deriving from `JsonNode` <br> - no checks in runtime <br> - IntelliSense | - a lot of additional methods <br> - does not accept a collection of different types | gives less possibilities than {1,2}, but requiers no additional checks |
+
+* Implicit operators for `JsonString`, `JsonBoolean` and `JsonNumber` as an additional feature.
+* `Sort` not implemented for `JsonArray`, beacuse there is no right way to compare `JsonObject`s. If user wants to sort `JsonArray` of `JsonNumber`s, `JsonBooleans`s or `JsonStrings` he/she now needs to do the following: convert `JsonArray` to regular array (by iterating through all elements), calling sort (and converting back to `JsonArray` if needed).
+* No support for duplicates of property names. Possibly, adding an option for user to choose from: "first value", "last value", or throw-on-duplicate.
+* Transformation API:
+    * `DeepCopy` method in JsonElement allowing to change JsonElement and JsonDocument into JsonNode recursively transforming all of the elements
+    * `AsJsonElement` method in JsonNode allowing to change JsonNode into JsonElement with IsImmutable property set to false
+    * `IsImmutable` property informing if JsonElement is keeping JsonDocument or JsonNode underneath
+    * `Parse(string)` in JsonNode to be able to parse Json string right into JsonNode if user knows he/she wants mutable version
+    * `DeepCopy` in JsonNode to make a copy of the whole tree
+    * `GetNode` and TryGetNode in JsonNode allowing to retrieve it from JsonElement
+
+## Implementation details
+* `JsonNumber` value is stored as a `string`.
+
+## Open questions
+API:
+* Do we want to add recursive equals on `JsonArray` and `JsonObject`?
+* Do we want to make `JsonNode`s derived types (and which) implement `IComparable`?
+* Would escaped characters be supported for creating `JsonNumber` from string? 
+* Is the API for `JsonNode` and `JsonElement` interactions sufficient? 
+* Do we want to support duplicate and order preservation/control when adding/removing values in `JsonArray`/`JsonObject`?
+* Do we want to have accelerator APIs for writing a JsonNode to a Utf8JsonWriter without having to go through JsonElement? (currently added as a `Parse` method in JsonNode)
+* Should JsonNode have a JsonValueKind property that a caller can inspect and cast to the right concrete type?
+* Let's say someone else passes a JsonNode to me, what can I do with it?
```

Comment:
That's an incredibly open question :).  Either there are scenarios for it, and they need to be called out, or there aren't, and the question is sort of moot...

Reply:
@ahsonkhan suggestion, but probably he maybe he didn't mean actually adding it, I'll remove it.

Reply:
Yep, this was meant as a question to provoke some thought as part of the API design of JsonNode and consider potential scenarios where one might run into that case (i.e. if JsonNode is used as an exchange type between two components to opaquely pass around a JSON payload). I would imagine specific/more meaningful questions would fall out of that exercise.